python/lsst/ts/dream/common/abstract_dream.py

```diff
@@ -0,0 +1,68 @@
+# This file is part of ts_dream.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["AbstractDream"]
+
+from abc import ABC, abstractmethod
+import typing
+
+
+class AbstractDream(ABC):
+    """Abstract class that defines the interface to communicate with a
+    DREAM instance."""
+
+    @abstractmethod
+    async def resume(self) -> None:
+        """Indicate that DREAM is permitted to resume automated operations."""
+        raise NotImplementedError
+
+    async def open_hatch(self) -> None:
+        """Open the hatch if DREAM has evaluated that it is safe to do so."""
+        raise NotImplementedError
+
+    async def close_hatch(self) -> None:
+        """Close the hatch."""
+        raise NotImplementedError
+
+    async def stop(self) -> None:
+        """Immediately stop operations and close the hatch."""
+        raise NotImplementedError
+
+    async def ready_for_data(self, ready: bool) -> None:
+        """Inform DREAM that Rubin Observatory is ready to receive data as
+        indicated.
+
+        Parameters
+        ----------
+        ready: `bool`
+            Rubin Observatory is ready to receive data (True) or not (False).
+        """
+        raise NotImplementedError
+
+    async def data_archived(self) -> None:
```

Comment:
So, two comments regarding this method. 

First, in general methods should start with a verb that express what the method does (`data` is a noun). I would recommend renaming it to `set_data_archived` or maybe `set_archived_data`.

Next, I am 95.02% certain that we will need to pass in some information to this method so DREAM know which data was archived. If we don't know yet what this data will be, we have two options;

1 - leave it like this but add a comment in the docstring that we have to establish the input.

2 - add a *args, **kwargs  argument and, once we figure out what we want, we make it explicit